### PR TITLE
Adding errors in s2d constructor

### PR DESCRIPTION
**Pull request by mughetto**

### CODE REVIEW

File: hbook/s2d.go [hunk 1]
```diff
@@ -35,11 +35,31 @@ func NewS2D(pts ...Point2D) *S2D {
 // NewS2DFrom creates a new 2-dim scatter with x,y data slices.
 //
 // It panics if the lengths of the 2 slices don't match.
-func NewS2DFrom(x, y []float64) *S2D {
+// Two first options are used to get eventual errors (with panic if sizes don't match)
+// Supports only symmetric errors
+func NewS2DFrom(x, y []float64, opts... []float64) *S2D {
```
Comment (sbinet) [inline hbook/s2d.go]: I think I'd rather split it into 2 (or 3) functions:

- `NewS2DFrom(x, y, []float64)` (unchanged)
- `NewS2DFromXYErrors(x, y, xerr, yerr []float64)`
- `NewS2DFromXYAsymmErrors(x,y,xerrlo,xerrhi,yerrlo,yerrhi []float64)`

what do you think ?

File: hbook/s2d.go [hunk 2]
```diff
@@ -35,11 +35,31 @@ func NewS2D(pts ...Point2D) *S2D {
...
+		}
+
+		if len(opts[0]) != len(x) {
+			panic("hbook: errors len mismatch data len")
+		}
+
+		xerr = opts[0]
+		yerr = opts[1]
+		
+		fmt.Println(xerr)
```
Comment (sbinet) [inline hbook/s2d.go]: remove

File: hbook/s2d.go [hunk 3]
```diff
@@ -35,11 +35,31 @@ func NewS2D(pts ...Point2D) *S2D {
...
+		fmt.Println(yerr)
```
Comment (sbinet) [inline hbook/s2d.go]: remove

File: hbook/s2d.go (hunk 1 above)
Comment (mughetto) [inline hbook/s2d.go]: Sure, I wasn't sure what was the spirit of Go regarding code duplication, but I'll write down explicitly the 3 functions you mentionned.

File: hbook/s2d.go (hunk 3 above)
Comment (mughetto) [inline hbook/s2d.go]: I guess you're targeting the stray printout right ;) ?

Comment (sbinet) [inline hbook/s2d.go]: yep.

File: hbook/s2d.go (hunk 1 above)
Comment (sbinet) [inline hbook/s2d.go]: you could have all these 3 functions redirect to a single one which takes all 6 slices but handles nil-slices as no input :)

Comment (mughetto) [inline hbook/s2d.go]: yep that's what I've started to wrork with :)

**Comment by sbinet** [comment]
@mughetto  gentle ping :)

**Comment by mughetto** [comment]
This diff set is not relevant anymore since you changed the API, gonna have fresh look now in a new fork and will make a new PR once ready.

**Pull request closed without merging**